### PR TITLE
catalog: Move builtin types to correct schema

### LINKED ISSUE #23833: We can resolve `mz_timestamp` from `pg_catalog`, but it should be `mz_catalog`
### What version of Materialize are you using?

62203e2d597983a404c958c8beae9fdaafbd74b6

### What is the issue?

```
        catalog
            .resolve_type(&PartialItemName {
                database: None,
                schema: Some("pg_catalog".to_string()), // todo: Why is this not `mz_catalog`?
                item: "mz_timestamp".to_string(),
            })
            .expect("mz_timestamp should exist");
```
see [here](https://materializeinc.slack.com/archives/C02FWJ94HME/p1702375729277789?thread_ts=1702328168.538389&cid=C02FWJ94HME).

[Here it is in context.](https://github.com/ggevay/materialize/blob/32289ce455864008d1ae1e8db28fc1cc05e60572/src/sql/src/pure.rs#L1503-L1509) Runnable with, e.g.,
```
bin/sqllogictest -- -v test/sqllogictest/materialized_views.slt
```

### PR DESCRIPTION
**Pull request by jkosh44**
Previously, we were accidentally putting all types in the pg_catalog schema. This commit moves the unsigned types and mz_timestamp to the mz_catalog schema and the acl_item type to the mz_internal schema.

Fixes #23833

### Motivation
This PR fixes a recognized bug.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [X] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [X] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - Move all non-PostgreSQL system types to the `mz_catalog` schema.

### CODE REVIEW

**Comment by jkosh44** [comment]
Note: DO NOT merge this until after the next release is cut.

File: src/sql/src/plan/statement.rs
Comment (benesch) [inline src/sql/src/plan/statement.rs:773]: Just to be safe, we should probably also add a case for `ty.oid() >= FIRST_USER_OID` and return an error.

File: test/sqllogictest/aclitem.slt
Comment (benesch) [inline test/sqllogictest/aclitem.slt:312]: These didn't _have_ to change, right? They're still visible via the default search path?

File: src/catalog/src/builtin.rs
Comment (benesch) [inline src/catalog/src/builtin.rs:6440]: Seems totally fine for the moment!

File: src/adapter/src/catalog/migrate.rs
Comment (benesch) [inline src/adapter/src/catalog/migrate.rs:325]: Ooh, nice, this is brilliant.

File: src/adapter/src/catalog/migrate.rs
Comment (benesch) [inline src/adapter/src/catalog/migrate.rs:96]: This missed v0.80.0, right?

File: src/adapter/src/catalog/open.rs
Comment (benesch) [inline src/adapter/src/catalog/open.rs:1208]: I see now what the problem was.

**Review by benesch — APPROVED** [review]
LGTM modulo naming of migration.

File: src/adapter/src/catalog/migrate.rs
Comment (jkosh44) [reply src/adapter/src/catalog/migrate.rs]: Yep fixed

File: test/sqllogictest/aclitem.slt
Comment (jkosh44) [reply test/sqllogictest/aclitem.slt:312]: No they didn't have to, they are still visible via the default search path.

Comment (benesch) [reply test/sqllogictest/aclitem.slt:312]: Great, just wanted to make sure I was understanding the change correctly.

Comment (jkosh44) [reply test/sqllogictest/aclitem.slt:312]: Quick sanity check:

```
materialize=> SELECT mz_internal.mz_aclexplode(array[null]::mz_aclitem[]);
ERROR:  MZ_ACL arrays must not contain null values
materialize=> SELECT mz_internal.mz_aclexplode(array[]::mz_aclitem[]);
 mz_aclexplode 
---------------
(0 rows)

```

**Review by benesch — APPROVED** [review]
LGTM!

File: src/sql/src/plan/statement.rs
Comment (jkosh44) [reply src/sql/src/plan/statement.rs:775]: Done

**Pull request merged**
